examples/ssd/train.py

```diff
@@ -0,0 +1,159 @@
+from chainercv.datasets import VOCDetectionDataset
+from chainercv.extensions import DetectionVOCEvaluator
+from chainercv.links.model.ssd import ConcatenatedDataset
+from chainercv.links.model.ssd import GradientScaling
+from chainercv.links.model.ssd import multibox_loss
+from chainercv.links import SSD300
+from chainercv import transforms
+
+from chainercv.links.model.ssd import random_crop_with_bbox_constraints
+from chainercv.links.model.ssd import random_distort
+from chainercv.links.model.ssd import resize_with_random_interpolation
+
+
+class MultiboxTrainChain(chainer.Chain):
+
+    def __init__(self, model, alpha=1, k=3):
+        super(MultiboxTrainChain, self).__init__(model=model)
+        self.alpha = alpha
+        self.k = k
+
+    def __call__(self, imgs, gt_mb_locs, gt_mb_labels):
+        mb_locs, mb_confs = self.model(imgs)
+        loc_loss, conf_loss = multibox_loss(
+            mb_locs, mb_confs, gt_mb_locs, gt_mb_labels, self.k)
+        loss = loc_loss * self.alpha + conf_loss
+
+        chainer.reporter.report(
+            {'loss': loss, 'loss/loc': loc_loss, 'loss/conf': conf_loss},
+            self)
+
+        return loss
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--batchsize', type=int, default=32)
+    parser.add_argument('--gpu', type=int, default=-1)
+    parser.add_argument('--out', default='result')
+    parser.add_argument('--resume')
+    args = parser.parse_args()
+
+    model = SSD300(
+        n_fg_class=len(voc_detection_label_names),
+        pretrained_model='imagenet')
+    model.use_preset('evaluate')
+    train_chain = MultiboxTrainChain(model)
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        model.to_gpu()
+
+    coder = copy.copy(model.coder)
+    coder.to_cpu()
+    size = model.insize
+    mean = model.mean
+
+    def transform(in_data):
+        img, bbox, label = in_data
+
+        img = random_distort(img)
+
+        if np.random.randint(2):
+            img, param = transforms.random_expand(
+                img, fill=mean, return_param=True)
+            bbox = transforms.translate_bbox(
+                bbox, y_offset=param['y_offset'], x_offset=param['x_offset'])
+
+        img, param = random_crop_with_bbox_constraints(
+            img, bbox, return_param=True)
+        bbox, param = transforms.crop_bbox(
+            bbox, y_slice=param['y_slice'], x_slice=param['x_slice'],
+            allow_outside_center=False, return_param=True)
+        label = label[param['index']]
+
+        _, H, W = img.shape
+        img = resize_with_random_interpolation(img, (size, size))
+        bbox = transforms.resize_bbox(bbox, (H, W), (size, size))
+
+        img, params = transforms.random_flip(
+            img, x_random=True, return_param=True)
+        bbox = transforms.flip_bbox(
+            bbox, (size, size), x_flip=params['x_flip'])
+
+        img -= np.array(mean)[:, np.newaxis, np.newaxis]
```

Comment:
> How do you use mean of this shape in resize_contain and random_expand?

Sorry, I understand how to use `(3, H, W)` mean in `resize_contain`.
But I think it is impossible to use `(3, H, W)` mean in `random_expand` because the size of output image changes randomly.

Anyway, I think supporting `(3, H, W)` can be another issue/PR.